src/writer/to_code.rs

```diff
@@ -116,12 +117,24 @@ impl ToCode for Chunk {
                 let s = format_block_one_line(&prefix, suffix, &value_strings, "", "");
                 vec![s]
             }
-            ErrorResultReturn { ref value } => {
+            AssertErrorSanity => {
+                let assert = format!(
+                    "assert_eq!(is_ok == {}, !error.is_null());",
+                    use_glib_type(env, "ffi::GFALSE")
+                );
```

Comment:
Maybe `assert_eq!(<bool as glib::translate::FromGlib>::from_glib(is_ok), error_is_null())` or so?

Reply:
I'm unconvinced.
This worked fine on ostree-rs and the CI is failing on a sign mismatch (`i32` vs `u32`).
I doubt further type hints would help, as it seems to be a case of `is_ok` not actually being a `gboolean`.
I think this hit some breakage further down in the line of consumers: for example `gtk_builder_add_from_resource` returns a `u32`, which is not the same type as a `gboolean`. From what I can see `FromGlib` do not (currently) cover that.

In which direction should we fix this?

Reply:
I think that function is not following the API guidelines by returning a `guint`, so having the implemented manually seems fine.